backend/src/main/controller/UserController.ts

```diff
@@ -0,0 +1,185 @@
+            return next(e);
+        }
+    };
+
+    // update one user
+    editUser = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the ID from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            // get parameters from the body
+            const userUpdateDTO: UserUpdateDTO = await UserUpdateValidation.schema.validateAsync(
+                req.body,
+                { abortEarly: false }
+            );
+            if (userUpdateDTO.password) {
+                userUpdateDTO.password = bcrypt.hashSync(userUpdateDTO.password, 8);
+            }
+            const result = await userRepository.update(id, userUpdateDTO);
+            res.status(204).send(result);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    // delete one user
+    deleteUser = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the ID from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            const result = await userRepository.delete(id);
+            return res.status(204).send(result);
+        } catch (e) {
+            next(e);
+        }
+    };
+
+    // get all saved promotion
+    getSaved = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id of user from the url
+            const id = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const userRepository = getCustomRepository(UserRepository);
+            const savedpromotions = await userRepository.findOneOrFail(id, {
+                relations: ["savedPromotions"]
+            });
+            
+            res.status(200).send(savedpromotions);
+        } catch (e) {
+            next(e);
+        }
+    };
+    // saved new promotion
+    newSaved = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            // get the id of user and promotion from the url
+            const uid = await IdValidation.schema.validateAsync(req.params.id, {
+                abortEarly: false,
+            });
+            const pid = await IdValidation.schema.validateAsync(req.params.pid, {
+                abortEarly: false,
+            });
+            // create new saved promotion for the user based on promotion id
+            const promotion = await getCustomRepository(PromotionRepository).findOneOrFail(pid);
+            const user = await getCustomRepository(UserRepository).findOneOrFail(uid);
+            const result = await getCustomRepository(UserRepository).addSavedPromotion(user, promotion);
```

Comment:
Hmm, do you know if there is any difference in doing it this way? I'm not sure if its faster since we don't need to load the whole User and Promotion object.

```suggestion
            const promotion = await getCustomRepository(PromotionRepository).hasId(pid);
            const user = await getCustomRepository(UserRepository).hasId(uid);
            if (promotion && user) {
              const savedPromotion = getCustomRepository(UserRepository).create({userid: uid, promotionId: pid})
              const result = await getCustomRepository(SavedPromotionRepository).save(savedpromotion)
            } else {
              // throw some error
            }
            
```